app/models/recipe.rb

```diff
@@ -1,3 +1,12 @@
 class Recipe < ApplicationRecord
-   validates_presence_of :name, :complexity, :genre
+    validates_presence_of :name, :complexity, :genre
+
+    has_many :recipe_ingredients
+    has_many :ingredients, through: :recipe_ingredients
+
+    def total_cost
+        ingredients.sum do |ingredient|
```

Comment:
Verrrrry close here, but this is a Ruby way of doing AR.... 
A red flag for you is anytime you're doing `... do |some_variable|`, that is a Ruby way of thinking. AR methods do not have the `do..` blocks. 

Here's how we could do this same thing in AR using `.sum`: 

```ruby
def total_cost
   ingredients.sum(:cost)
end
```

We still use the `ingredients` association because we're in an instance method, but the `.sum` method takes an argument of a column on the table it's called on (so, the `cost` column on the `ingredients` table) and adds them together for us _in the database_. Databases are tuned to do calculations & sorting way faster than other languages, so let them do the hard work for us.